homeassistant/components/homekit/accessories.py

```diff
@@ -237,6 +239,36 @@ def setup_message(self):
         """Prevent print of pyhap setup message to terminal."""
         pass
 
+    def get_snapshot(self, info):
+        """Get snapshot from accessory if supported."""
+        acc = self.accessories.get(info["aid"])
+        if acc is None:
+            raise ValueError("Requested snapshot for missing accessory")
+        if not hasattr(acc, "get_snapshot"):
+            raise ValueError(
+                "Got a request for snapshot, but the Accessory "
+                'does not define a "get_snapshot" method'
+            )
+        return acc.get_snapshot(info)
+
+
+class HomeServerHandler(HAPServerHandler):
```

Comment:
`HAPServerHandler` upstream seems to already implement exactly this, can we get rid of this class entirely or have i missed something?

Reply:
Ah, acc.get_snapshot returns an `Image` and not bytes. But we can make `get_snapshot` return bytes and then the upstream server handler will work?

Yours looks like it sets the content type based on the `Image` too and upstream can't do that. But the public HAP spec says the snapshot must be JPEG, so maybe that's not a problem?

Reply:
I duplicated this to handle content-type from `camera.async_get_image()`. I wasn't sure if it's always jpeg for all camera implementations. If it is, we can ditch it. If not, it should probably just error if `async_get_image()` doesn't return a jpeg since the spec demands it.